etc/generate-package-yml-files-from-PackageInfo-files.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+# How to use this:
+# - Edit the gapTarballURL variable. Point it to the tarball of a gap release.
+# - run this script from the `GapWWW` repo root directory via
+#   `./etc/generate-package-yml-files-from-PackageInfo-files.sh`
+#
+# An executable `gap` must be in your $PATH, see the line before the last one
+# of this script.
+
+gapTarballURL='https://files.gap-system.org/gap-4.11/tar.gz/gap-4.11.0.tar.gz'
+collectionDir='_Packages'
+mkdir ${collectionDir}/tmp
+cd ${collectionDir}/tmp
+echo "----    Getting the GAP tarball    ----\n"
+wget ${gapTarballURL}
+echo "----    Unpacking the GAP tarball    ----\n"
+tar -xf gap-4.11.0.tar.gz
+echo "----    Generating YML files    ----\n"
+mv gap-*/pkg .
+rm -rf gap-*
+cd ../..
+find ${collectionDir}/tmp/pkg/ -name 'PackageInfo.g' > ${collectionDir}/tmp/list-of-paths-to-packageinfo-files.txt
+# TODO use the gap version from the tar ball?
+gap etc/generate-package-yml-files-from-PackageInfo-files.g
+rm -rf ${collectionDir}/tmp
```

Comment:
Hmm, that's a bit odd. I would do one of these:
1. Let the GAP script only handle a single package at a time, and instead loop over the packages in this script. Something like this (untested, just meant to illustrate)
```bash
export wwwdir =$PWD
for f in $(find ${pkgdir} -name PackageInfo.g) ; do
  ( cd $(dirname ${pkgdir}) && gap ${wwwdir}/etc/generate-package-yml-files-from-PackageInfo-files.g )
done
```
Here the GAP script could extract the target directory from the env var `wwwdir`.

2. Let GAP take care of iterating over the files ; use `DirectoryContents` for that.


In both cases, you can pass information (such as the `pkgdir` as well as the `wwwdir` to GAP either via environment variables; or by using the `-c` command line option, e.g.:
```
gap -c "pkgdir := \"${pkgdir}\"; wwwdir := \"${wwwdir}\";" FILENAME.g
```
or if FILENAME only contains functions:
```
gap  FILENAME.g -c "UpdatePackages(\"${pkgdir}\", \"${wwwdir}\"); QUIT"
```

Reply:
1. I didn't want to have to start up a separate gap for each Package, since on my machine that takes an extra five minutes. Right now, if I only run the last lines of this script it takes less than 10 seconds on my machine. Besides from not creating a file on the file system, are there advantages of the code you wrote? Is it more portable or something? I could use the `for f in $(find ${pkgdir} -name PackageInfo.g) ; do` loop to create a string from which in turn gap can turn into a list of strings.
1. I found using `find` quicker, in the sense of time to write the code, than writing a gap function which recurses over the directory.

I'll update the script to use a parameter and use -c to pass that to gap.

Reply:
@fingolfin I've updated the shell script and the corresponding gap file.

Reply:
updating a few more things..

Reply:
I've added error messages and cleaned the script up a little.